sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/GenericArrayData.scala

```diff
@@ -122,7 +122,7 @@ class GenericArrayData(val array: Array[Any]) extends ArrayData {
             if (!o2.isInstanceOf[Double] || ! java.lang.Double.isNaN(o2.asInstanceOf[Double])) {
               return false
             }
-          case _ => if (o1 != o2) {
+          case _ => if (!o1.equals(o2)) {
```

Comment:
Are there any needs to handle `Array[Byte]` separately above?

Reply:
in java `byte[]` or other primitive arrays doesn't have a proper `equals` implementation.
```
scala> Array(1) == Array(1)
res0: Boolean = false
```